sycl-fusion/passes/kernel-info/SYCLKernelInfo.cpp

```diff
@@ -49,13 +53,21 @@ void SYCLModuleInfoAnalysis::loadModuleInfoFromMetadata(Module &M) {
 
     // Operand 0: Kernel name
     auto Name = cast<MDString>(*It)->getString().str();
-    SYCLKernelInfo KernelInfo{std::move(Name)};
     ++It;
 
     // Operand 1: Argument kinds
     auto *ArgsKindsMD = cast<MDNode>(*It);
+    ++It;
+
+    // Operand 2: Argument usage mask
+    auto *ArgsUsageMaskMD = cast<MDNode>(*It);
+    ++It;
+
+    assert(ArgsKindsMD->getNumOperands() == ArgsUsageMaskMD->getNumOperands());
+    SYCLKernelInfo KernelInfo{std::move(Name), ArgsKindsMD->getNumOperands()};
```

Comment:
We're copying the string anyway, so...
```suggestion
    SYCLKernelInfo KernelInfo{Name, ArgsKindsMD->getNumOperands()};
```